src/Adapter/Product/Image/Repository/ProductImageMultiShopRepository.php

```diff
@@ -506,4 +541,37 @@ public function getImageById(ImageId $imageId): Image
 
         return $image;
     }
+
+    /**
+     * @return ImageType[]
+     */
+    public function getProductImageTypes(): array
+    {
+        try {
+            $results = ImageType::getImagesTypes('products');
+        } catch (PrestaShopException $e) {
+            throw new CoreException('Error occurred when trying to get product image types');
+        }
+
+        if (!$results) {
+            return [];
+        }
+
+        $imageTypes = [];
+        foreach ($results as $result) {
+            $imageType = new ImageType();
+            $imageType->id = (int) $result['id_image_type'];
+            $imageType->name = $result['name'];
+            $imageType->width = (int) $result['width'];
+            $imageType->height = (int) $result['height'];
+            $imageType->products = (bool) $result['products'];
+            $imageType->categories = (bool) $result['categories'];
+            $imageType->manufacturers = (bool) $result['manufacturers'];
+            $imageType->suppliers = (bool) $result['suppliers'];
+            $imageType->stores = (bool) $result['stores'];
+            $imageTypes[] = $imageType;
```

Comment:
```suggestion
            $imageType = new ImageType();
            $imageType->hydrate($result);
            $imageTypes[] = $imageType;
```

maybe you could do this instead, although I'm not sure why you need to return array of objects

Reply:
Actually, I just copied this from the original `ProductImageRepository` service, I didn't think about what I copied since the purpose of this PR is not to refacto this part, although I think returning object makes sense I'm not sure it should be some ObjectModel, and the way they are created can definitely be improved but again not the point here

You're not the first one to point at stuff about this method and the fact that I turned this service as public, if it really bothers you all I can just revert this change as it's not mandatory for this PR and I don't want it to be blocked for parallel issues

Reply:
Regarding the getRows insertBulk comment, we already discussed this with @FabienPapet I agree this could be useful in a generic component but again not the purpose here, so if a refacto aims at placing this in a component it should be thought out to see where it should be

For now these are just helper methods, they are **private** so not at all usable outside of this class, so they have no impact outside that's why I thought it's ok for now And again I don't want the PR being blocked for over engineering a component I added locally just to help myself and clean the code a bit 😅

Reply:
Sure, I understand :) Like I wrote: non-blocking comment. Do you think we can make sure we note these improvements somewhere?  Maybe a `@todo` inside the code (could be in other PR even focused on marking things to refacto).

Reply:
Sure, we have a refacto that aims at merging `ProductImageRepository` with `ProductImageMultiShopRepository` so I'll try and improve it or at least add the todo at that moment

Reply:
noted.
Related pr https://github.com/PrestaShop/PrestaShop/pull/31611